core/include/eigen/include/algebra/definitions/eigen.hpp

```diff
@@ -199,6 +199,18 @@ namespace algebra
                 return (_data.isApprox(rhs._data));
             }
 
+            /** Rotate a vector into / from a frame 
+             * 
+             * @param m is the rotation matrix
+             * @param v is the vector to be rotated
+             */
+	    
+            static vector3 rotate(const Eigen::Transform<scalar, 3, Eigen::Affine> &m, const vector3 &v)
```

Comment:
Maybe we can try this with the MatrixBase type and Eigen::Block<Derived>()? Then it might work with expressions, too

Reply:
You mean changing `Eigen::Transform` into the type of `matrix44` in the argument? If so, that will lead into different argument semantic between eigen and other plugins. The argument type for `m` input should be same with the type of `_data`

Reply:
No, if this works, its fine, thank you!